tests/test_2408_layoutbuilder_in_numba.py

```diff
@@ -0,0 +1,1462 @@
+
+    builder = lb.Record(
+        [
+            lb.Numpy(np.float64),
+            lb.Numpy(np.int64),
+            lb.Numpy(np.uint8),
+        ],
+        ["one", "two", "three"],
+    )
+    assert f4(builder) == 0
+
+    builder = lb.Tuple(
+        [lb.Numpy(np.float64), lb.ListOffset(np.int64, lb.Numpy(np.int32))]
+    )
+    assert f4(builder) == 0
+
+    builder = lb.Union(
+        np.int8,
+        np.int64,
+        [
+            lb.Numpy(np.float64),
+            lb.ListOffset(np.int64, lb.Numpy(np.int32)),
+        ],
+    )
+    assert f4(builder) == 0
+
+
+def test_Numpy_from_buffer():
+    @numba.njit
+    def f5(debug=True):
+        growablebuffer = ak.numba.GrowableBuffer(np.float64)
+        growablebuffer.append(66.6)
+        growablebuffer.append(77.7)
+        return growablebuffer
+
+    out = f5()
+    assert out.snapshot().tolist() == [66.6, 77.7]
+
+    @numba.njit
+    def f6():
+        growablebuffer = ak.numba.GrowableBuffer(np.float64)
+        growablebuffer.append(66.6)
+        growablebuffer.append(77.7)
+
+        return lb._from_buffer(growablebuffer)
+
+    out = f6()
+    assert isinstance(out, lb.Numpy)
+    assert out.dtype == np.dtype(np.float64)
+    assert len(out) == 2
+
+    assert ak.to_list(out.snapshot()) == [66.6, 77.7]
+
+
+def test_Numpy_ctor():
+    @numba.njit
+    def f7():
+        return lb.Numpy("f4")
+
+    out = f7()
+    assert isinstance(out, lb.Numpy)
+    assert out.dtype == np.dtype("f4")
+    assert len(out) == 0
+
+    @numba.njit
+    def f8():
+        return lb.Numpy(np.float32)
+
+    out = f8()
+    assert isinstance(out, lb.Numpy)
+    assert out.dtype == np.dtype(np.float32)
+    assert len(out) == 0
+
+    @numba.njit
+    def f9():
+        return lb.Numpy(np.dtype(np.float32))
+
+    out = f9()
+    assert isinstance(out, lb.Numpy)
+    assert out.dtype == np.dtype(np.float32)
+    assert len(out) == 0
+
+
+def test_Numpy_append():
+    @numba.njit
+    def f10(builder):
+        for i in range(8):
+            builder.append(i)
+
+    builder = lb.Numpy(np.float32)
+
+    f10(builder)
+
+    assert ak.to_list(builder.snapshot()) == list(range(8))
+
+    f10(builder)
+
+    assert ak.to_list(builder.snapshot()) == list(range(8)) + list(range(8))
+
+
+def test_Numpy_extend():
+    @numba.njit
+    def f11(builder):
+        builder.extend(np.arange(8))
+
+    builder = lb.Numpy(np.float32)
+
+    f11(builder)
+
+    assert ak.to_list(builder.snapshot()) == list(range(8))
+
+    f11(builder)
+
+    assert ak.to_list(builder.snapshot()) == list(range(8)) + list(range(8))
+
+
+def test_Numpy_snapshot():
+    @numba.njit
+    def f12(builder):
+        return builder.snapshot()
+
+    builder = lb.Numpy(np.float32)
+
+    assert ak.to_list(f12(builder)) == []
+
+    builder.extend(range(8))
+
+    assert ak.to_list(f12(builder)) == list(range(8))
+
+    builder.extend(range(8))
+
+    assert ak.to_list(f12(builder)) == list(range(8)) + list(range(8))
+
+
```

Comment:
It doesn't need to be possible to do a `snapshot` inside of the compiled context. And if you did, the memory that the `ArrayView` points to (which is what an `ak.Array` is in a compiled context) couldn't be owned by anything, so there's no way to even do it without changing our ownership model.

```suggestion
```

Reply:
done